### PR TITLE
OADP-171 Adding ability for the default secrets update, to queue all DPA's

### PR DESCRIPTION
**Pull request by shawn-hurley**
Want to see what E2E breaks, if any before adding another test case.

[OADP-171](https://issues.redhat.com//browse/OADP-171)

### CODE REVIEW

File: controllers/dpa_controller.go [hunk 1]
```diff
@@ -146,16 +148,53 @@ func (r *DPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+	l.mu.RUnlock()
+	if ok {
+		list := &oadpv1alpha1.DataProtectionApplicationList{}
+		err := l.client.List(context.Background(), list)
+		if err != nil {
+			//TODO: There should be a log here, and we probaly want to somehow re-queue the secret
+			return
+		}
+		for _, dpa := range list.Items {
+			q.Add(reconcile.Request{
```
Comment (savitharaghunathan) [inline controllers/dpa_controller.go:188]: @shawn-hurley this is probably a dumb question. Are we not checking for labels anymore?

File: controllers/dpa_controller.go [hunk 2]
```diff
@@ -146,16 +148,53 @@ func (r *DPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
 		Owns(&corev1.ConfigMap{}).
-		Watches(&source.Kind{Type: &corev1.Secret{}}, &labelHandler{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, &secretHandler{}).
 		WithEventFilter(veleroPredicate(r.Scheme)).
 		Complete(r)
 }
 
-type labelHandler struct {
+type secretHandler struct {
+	defauldSecrets map[string]*struct{}
```
Comment (kaovilai) [inline controllers/dpa_controller.go:157]: Gotta love this guy.
```suggestion
	defaultSecrets map[string]*struct{}
```

File: controllers/dpa_controller.go [hunk 3]
```diff
@@ -146,16 +148,53 @@ func (r *DPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-func (l *labelHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+func createSecretHandler() *secretHandler {
+	defaultSecrets := map[string]*struct{}{}
+	for _, fields := range credentials.PluginSpecificFields {
+		if fields.IsCloudProvider {
+			defaultSecrets[fields.SecretName] = nil
+		}
+	}
+	return &secretHandler{
+		defauldSecrets: defaultSecrets,
```
Comment (kaovilai) [inline controllers/dpa_controller.go:170]:
```suggestion
		defaultSecrets: defaultSecrets,
```

File: controllers/dpa_controller.go [hunk 4]
```diff
@@ -146,16 +148,53 @@ func (r *DPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+		mu:             sync.RWMutex{},
+	}
+}
+
+func (l *secretHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
 	// check for the label & add it to the queue
+	l.mu.RLock()
+	_, ok := l.defauldSecrets[evt.Object.GetName()]
```
Comment (kaovilai) [inline controllers/dpa_controller.go:178]:
```suggestion
	_, ok := l.defaultSecrets[evt.Object.GetName()]
```

File: controllers/dpa_controller.go [hunk 5]
```diff
@@ -168,8 +207,29 @@ func (l *labelHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInt
 	}})
 
 }
-func (l *labelHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+func (l *secretHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
 
+	l.mu.RLock()
+	_, ok := l.defauldSecrets[evt.Object.GetName()]
```
Comment (kaovilai) [inline controllers/dpa_controller.go:213]:
```suggestion
	_, ok := l.defaultSecrets[evt.Object.GetName()]
```

File: controllers/dpa_controller.go [hunk 6]
```diff
@@ -181,7 +241,28 @@ func (l *labelHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInt
 	}})
 
 }
-func (l *labelHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+func (l *secretHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	l.mu.RLock()
+	_, ok := l.defauldSecrets[evt.ObjectNew.GetName()]
```
Comment (kaovilai) [inline controllers/dpa_controller.go:246]:
```suggestion
	_, ok := l.defaultSecrets[evt.ObjectNew.GetName()]
```

File: controllers/dpa_controller.go [hunk 7]
```diff
@@ -193,8 +274,29 @@ func (l *labelHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInt
 	}})
 
 }
-func (l *labelHandler) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+func (l *secretHandler) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
 
+	l.mu.RLock()
+	_, ok := l.defauldSecrets[evt.Object.GetName()]
```
Comment (kaovilai) [inline controllers/dpa_controller.go:280]:
```suggestion
	_, ok := l.defaultSecrets[evt.Object.GetName()]
```

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: controllers/dpa_controller.go (hunk 1 above)
Comment (shawn-hurley) [reply controllers/dpa_controller.go:188]: The default secrets are hardcoded names, if we are watching all the secrets, lets just use the names was my thought

**Review by savitharaghunathan — APPROVED** [review]
/lgtm. Thanks @shawn-hurley

**Comment by kaovilai** [comment]
@shawn-hurley are you adding another test case?

**Comment by shawn-hurley** [comment]
I might need a hand, but I can try and get that moving. What is the time frame that we need to get this in? 

@weshayutin @kaovilai

**Comment by kaovilai** [comment]
@shawn-hurley moved to 1.1.2 so we have some time in the next sprint.

**Pull request closed without merging**